src/Symfony/Component/Form/Extension/DataCollector/FormDataExtractor.php

```diff
@@ -199,4 +200,34 @@ private function buildId(FormInterface $form)
 
         return $id;
     }
+
+    /**
+     * Converts the variable into a serializable Data instance.
+     *
+     * This array can be displayed in the template using
+     * the VarDumper component.
+     *
+     * @param mixed $var
+     *
+     * @return Data
+     */
+    private function cloneVar($var)
+    {
+        if (null === $this->cloner) {
+            $this->cloner = new VarCloner();
+            $this->cloner->addCasters(array(
+                \Exception::class => function (\Exception $e, array $a, Stub $s) { unset($a["\0Exception\0previous"]); ++$s->cut; return $a; },
+                FormInterface::class => function (FormInterface $f, array $a, Stub $s) { return StubCaster::cutInternals($f, $a, $s, true); },
```

Comment:
does this line avoid dumping the Form in the constraint violations? If it does, I'm not sure if we should include `root` at all, it only shows `Form {#986 …13}` now, not that usefull.